src/Utilities/functions.jl

```diff
@@ -379,16 +372,28 @@ then `isapprox_zero(f)` is `false` but `isapprox_zero(MOIU.canonical(f))` is
 """
 function isapprox_zero end
 
-function isapprox_zero(f::MOI.ScalarAffineFunction, tol)
-    isapprox_zero(f.constant, tol) && all(t -> isapprox_zero(t, tol),
-                                           f.terms)
+isapprox_zero(α::AbstractFloat, tol) = -tol <= α <= tol
+isapprox_zero(α::Union{Integer, Rational}, tol) = iszero(α)
+function isapprox_zero(f::MOI.AbstractFunction, tol)
+    return _iszero(f, α -> isapprox_zero(α, tol))
+end
+
+function Base.iszero(f::Union{MOI.ScalarAffineFunction{T},
+                              MOI.ScalarQuadraticFunction{T}}) where T
+    return _iszero(canonical(f), iszero)
+end
+
+function _iszero(f::MOI.ScalarAffineFunction, _is_zero::Function)
```

Comment:
Please improve the naming :)
Local variables shouldn't begin with underscores. And the distinction between `_iszero` and `_is_zero` is confusing.